repair/repair.cc

```diff
@@ -950,9 +953,9 @@ future<> shard_repair_task_impl::do_repair_ranges() {
         // repair all the ranges in limited parallelism
         rlogger.info("repair[{}]: Started to repair {} out of {} tables in keyspace={}, table={}, table_id={}, repair_reason={}",
                 id.uuid(), idx + 1, table_ids.size(), _status.keyspace, table_name, table_id, _reason);
-        co_await coroutine::parallel_for_each(ranges, [this, table_id] (auto&& range) {
-            return with_semaphore(rs.get_repair_module().range_parallelism_semaphore(), 1, [this, &range, table_id] {
-                return repair_range(range, table_id).then([this] {
+        co_await coroutine::parallel_for_each(ranges, [this, table_id, table_name] (auto&& range) {
```

Comment:
since this runs in a coroutine, we can capture `&table_name` by reference and copy it only when calling `repair_range`.